database/admin_views.py

```diff
@@ -0,0 +1,91 @@
+import json
+
+from flask import redirect, request, url_for
+from flask_login import current_user
+from flask_admin import Admin
+from flask_admin.contrib.sqla import ModelView
+from flask_admin.form import fields
+
+from .models import db, User, Person, Role, AccessLevel
+from .events import Event, EventRegistration
+
+
+class PrettyJSONField(fields.JSONField):
+    def _value(self):
+        if self.raw_data:
+            return self.raw_data[0]
+        elif self.data:
+            return json.dumps(self.data, ensure_ascii=False, indent=2)
+        else:
+            return ''
+
+
+class AuthorizedView(ModelView):
+    access_level = None
+
+    def is_accessible(self):
+        if current_user is not None:
+            return current_user.has_access(self.access_level)
+
+    def inaccessible_callback(self, name, **kwargs):
```

Comment:
Mhhh this will effectively redirect to the index page, which is probably not very useful. Changing this to `abort(401)` so a user will know what's the problem.